docs/pages/contributing/instantiations.md

```diff
@@ -0,0 +1,66 @@
+# Instantiating Classes in SparseBase
+
+While SparseBase supports a header-only mode, our main focus is in the compiled version of the library. 
+This is mainly due to the large impact of header-only mode on the build times of user's projects.
+
+As a result of this, we try to keep the header files as light as possible by avoiding putting implementations there.
+There are a few exceptions to this (like the `As<>()` function of `Format`) 
+but this is only done in very rare situations.
+
+Hence, any templated class or function you create in SparseBase will require instantiation as it needs to be compiled
+with the library with the types known at compile time. This is achieved with the script 
+`src/generate_explicit_instantiations.py`.
+
+> Note that this script is not intended to be called manually besides debugging. It will be called automatically
```

Comment:
For some reason this quote didn't show up on my computer. Just looks indented.
![resim](https://user-images.githubusercontent.com/11246827/193450096-5541e2b9-ea13-4fed-8bff-fde5ad65c867.png)

Reply:
Same on mine, I don't think we can fix that :(